analyzer/src/main/kotlin/managers/Gradle.kt

```diff
@@ -164,6 +171,16 @@ class Gradle(
             gradleConnector.daemonMaxIdleTime(10, TimeUnit.SECONDS)
         }
 
+        // Gradle's default maximum heap is 512 MiB which is too low for bigger projects,
+        // see https://docs.gradle.org/current/userguide/build_environment.html#sec:configuring_jvm_memory.
+        // Set the value to empirically determined 8 GiB if no value is set in "~/.gradle/gradle.properties".
+        val jvmArgs = gradleProperties.find { (key, _) ->
+            key == "org.gradle.jvmargs"
+        }?.second?.split(" ")?.toMutableList() ?: mutableListOf()
+        if (!jvmArgs.any { it.contains(other = "-xmx", ignoreCase = true) }) {
```

Comment:
While I appreciate the effort, is searching the individual " "-separated pieces really much safer than just searching the whole string for "-xmx", given that it's quite unlikely that it contains strings like "foo-xmx"? I'm fine either way, just asking for your opinion whether we could simplify the code.

Reply:
> is searching the individual " "-separated pieces really much safer than just searching the whole string for "-xmx"

While I don't think `foo-xmx` will be used by anyone, I'd like to keep it the way I wrote it (with the simplification you mentioned) because the alternative would be this:
```kt
var jvmArgs = gradleProperties.find { (key, _) ->
    key == "org.gradle.jvmargs"
}?.second ?: ""

if (!jvmArgs.contains(other = "-xmx", ignoreCase = false)) {
    jvmArgs += " -Xmx8g"
}
```
Where I don't like the mutable jvmArgs and the required space in `" -Xmx8g"`. As `addJvmArguments` allows varargs I think the implementation with the splitting of the arguments is "cleaner"